### PR TITLE
Add cluster state

### PR DESCRIPTION
**Pull request by stgleb**
Track cluster and node state changes in separate
goroutine.

### CODE REVIEW

File: pkg/node/node.go [hunk 1]
```diff
@@ -6,21 +6,36 @@ import (
...
+
+const (
+	StatePlanned      NodeState = "planned"
+	StateBuilding     NodeState = "building"
+	StateProvisioning NodeState = "provisioning"
+	StateError        NodeState = "error"
+	StateActive       NodeState = "active"
+
+	RoleMaster Role = "master"
+	RoleNode   Role = "node"
```
Comment (saheienko) [inline pkg/node/node.go]: Probably, it should be `worker`.

File: pkg/node/node.go [hunk 2]
```diff
@@ -6,21 +6,36 @@ import (
...
 
+type NodeState string
+
+type Role string
+
+const (
+	StatePlanned      NodeState = "planned"
+	StateBuilding     NodeState = "building"
+	StateProvisioning NodeState = "provisioning"
+	StateError        NodeState = "error"
```
Comment (saheienko) [inline pkg/node/node.go]: Does an `error` state is self descriptive?

File: pkg/provisioner/provisioner_test.go [hunk 3]
```diff
@@ -16,6 +16,7 @@ import (
 	"github.com/supergiant/supergiant/pkg/testutils"
 	"github.com/supergiant/supergiant/pkg/workflows"
 	"github.com/supergiant/supergiant/pkg/workflows/steps"
+	"time"
```
Comment (saheienko) [inline pkg/provisioner/provisioner_test.go]: Please, run `make fmt`.

File: pkg/node/node.go (hunk 2 above)
Comment (stgleb) [reply pkg/node/node.go]: Not that much, more descriptive error is stored in corresponding task.

File: pkg/provisioner/provisioner_test.go (hunk 3 above)
Comment (stgleb) [reply pkg/provisioner/provisioner_test.go]: my fault, even though`make fmt` doesn't heal this

File: pkg/node/node.go (hunk 1 above)
Comment (stgleb) [reply pkg/node/node.go]: I think we can discuss and set up naming convention for entities while we are all here, I am open to any propositions.

**Pull request merged**